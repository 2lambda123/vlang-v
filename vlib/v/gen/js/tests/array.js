// V_COMMIT_HASH 0de70e8
// V_CURRENT_COMMIT_HASH 4271eb4
// Generated by the V compiler

"use strict";

/** @namespace builtin */
const builtin = (function () {
	/**
	 * @param {any} s
	 * @returns {void}
	 * @function
	*/
	function println(s) {
		console.log(s);
	}
	
	/**
	 * @param {any} s
	 * @returns {void}
	 * @function
	*/
	function print(s) {
		process.stdout.write(s);
	}

	/* module exports */
	return {
		println,
		print,
	};
})();

/** @namespace main */
const main = (function () {
	/**
	 * @param {...number} args
	 * @returns {void}
	 * @function
	*/
	function variadic(...args) {
		builtin.println(args);
		builtin.println(args[0]);
		builtin.println(args[1]);
	}
	
	/**
	 * @returns {void}
	 * @function
	*/
	function vararg_test() {
		variadic(1, 2, 3);
	}
	
	/* program entry point */
	(function() {
		vararg_test();
		/** @type {string[]} - arr1 */
		const arr1 = ["Hello", "JS", "Backend"];
		/** @type {number[]} - arr2 */
		let arr2 = [1, 2, 3, 4, 5];
		/** @type {string[]} - slice1 */
		const slice1 = arr1.slice(1, 3);
		/** @type {number[]} - slice2 */
		const slice2 = arr2.slice(0, 3);
		/** @type {number[]} - slice3 */
		const slice3 = arr2.slice(3, arr2.length);
		/** @type {string} - idx1 */
		const idx1 = slice1[1];
		arr2[0] = 1;
		arr2[0 + 1] = 2;
		builtin.println(arr2);
		/** @type {string} - slice4 */
		const slice4 = idx1.slice(0, 4);
		builtin.println(slice4);
		/** @type {byte} - idx2 */
		const idx2 = slice4[0];
		/** @type {Map<string, string>} - m */
		let m = new Map();
		/** @type {string} - key */
		const key = "key";
		m[key] = "value";
		/** @type {string} - val */
		const val = m["key"];
		builtin.println(val);
	})();

	/* module exports */
	return {};
})();


