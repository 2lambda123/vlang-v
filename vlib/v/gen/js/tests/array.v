fn variadic(args ...int) {
	println(args)
	println(args[0])
	println(args[1])
}

fn vararg_test() {
	variadic(1, 2, 3)
}

vararg_test()

arr1 := ['Hello', 'JS', 'Backend']
mut arr2 := [1, 2, 3, 4, 5]

// Array slices
slice1 := arr1[1..3]
slice2 := arr2[..3]
slice3 := arr2[3..]

// Array indexes
idx1 := slice1[1]
arr2[0] = 1
arr2[0 + 1] = 2
println(arr2)

// String slices
slice4 := idx1[..4]
println(slice4) // 'Back'

// String indexes
idx2 := slice4[0]

// TODO: This does not work for now
// arr2[0..1] = arr2[3..4]
// println(arr2)


// Maps
mut m := map[string]string
key := 'key'
m[key] = 'value'
val := m['key']
println(val)