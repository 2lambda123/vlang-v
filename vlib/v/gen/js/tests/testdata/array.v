fn main() {
	mut arr := [8, 7, 6, 5, 4, 3, 2, 1, 0, 0, 0]
	arr.sort()
	println(arr)
	arr[9] = 100
	arr[10] = 120
	arr.sort()

	println(arr)

	println([1, 2, 3, 4].repeat(2))

	println(arr.index(120))

	println([1, 2, 3, 4, 5].join('=>'))

	res := arr[4..7]

	println(res)
}
