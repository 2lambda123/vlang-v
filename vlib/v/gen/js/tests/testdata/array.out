[0,0,0,1,2,3,4,5,6,7,8]
[0,0,0,1,2,3,4,5,6,100,120]
[1,2,3,4,1,2,3,4,1,2,3,4,1,2,3,4,1,2,3,4,1,2,3,4,1,2,3,4,1,2,3,4]
10
1=>2=>3=>4=>5
[2,3,4]